tensorflow/python/keras/_impl/keras/estimator.py

```diff
@@ -49,6 +50,8 @@
 def _cast_tensor_to_floatx(x):
   """Cast tensor to keras's floatx dtype if it is not already the same dtype."""
   if x.dtype == K.floatx():
+    if not check_ops.is_numeric_tensor(x):
+      return sparse_tensor_lib.convert_to_tensor_or_sparse_tensor(x)
```

Comment:
Could you add a comment to explain what this is for? In which cases would we have `x.dtype == K.floatx() and not check_ops.is_numeric_tensor(x)`? If x is a Numpy array?